google-cloud-debugger/docs/instrumentation.md

```diff
@@ -0,0 +1,136 @@
+# Stackdriver Debugger Instrumentation
+
+Stackdriver Debugger is a feature of the Google Cloud Platform that lets
+you inspect the state of an application at any code location without using
+logging statements and without stopping or slowing down your applications.
+Your users are not impacted during debugging. Using the production
+debugger you can capture the local variables and call stack and link it
+back to a specific line location in your source code. You can use this to
+analyze the production state of your application and understand the
+behavior of your code in production.
+
+For general information about Stackdriver Debugger, read [Stackdriver
+Debugger Documentation](https://cloud.google.com/debugger/docs/).
+
+The Stackdriver Debugger Ruby library, `google-cloud-debugger`, provides an
+easy-to-use debugger instrumentation that reports state data, such as
+value of program variables and the call stack, to Stackdriver Debugger
+when the code at a breakpoint location is executed in your Ruby
+application. See the [instrumenting your app](#instrumenting-your-app)
+section for how to debug your application, in both development and production.
+
+## Instrumenting Your App
+
+This instrumentation library provides the following features to help you
+debug your applications in production:
+
+*   Automatic application registration. It facilitates multiple running
+    instances of same version of application when hosted in production.
+*   A background debugger agent that runs side-by-side with your
+    application that automatically collects state data when code is
+    executed at breakpoint locations.
+*   A Rack middleware and Railtie that automatically manages the debugger
+    agent for Ruby on Rails and other Rack-based Ruby applications.
+
+Other Stackdriver Debugger service features not covered by this
```

Comment:
The goal is to emphasis the debugger agent is only part of the debugging process. There are additional steps users need to take in order to get it working. Maybe I should just say this directly, cause it's not documented anywhere else (like many other things in GCP land).